docs/addons/ssm-agent.md

```diff
@@ -0,0 +1,96 @@
+# SSM Agent Add-on
+
+This add-on uses the Kubernetes DaemonSet resource type to install AWS Systems Manager Agent (SSM Agent) on all worker nodes, instead of installing it manually or replacing the Amazon Machine Image (AMI) for the nodes. DaemonSet uses a CronJob on the worker node to schedule the installation of SSM Agent.
+
+A common use-case for installing SSM Agent on the worker nodes is to be able open a terminal session on an instance without the need to create a bastion instance and without having to install SSH keys on the worker nodes.
+
+The AWS Identity and Access Management (IAM) managed role **AmazonSSMManagedInstanceCore** provides the required permissions for SSM Agent to run on EC2 instances. This role is automatically attached to the instances when this add-on is enabled.
+
+**Limitations**
+
+* This add-on isn't applicable to AWS Fargate, because DaemonSets aren't supported on the Fargate platform.
```

Comment:
Just a reminder for myself: we need a way to mark add-ons as incompatible with cluster providers, e.g. this one should be incompatible with Fargate / Bottlerocket.

Reply:
Ticket here. https://github.com/aws-quickstart/quickstart-ssp-amazon-eks/issues/134